Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +7,217 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            
+
+            var firstAccount = peopleAndBalances[0].Split(", ");
+            var firstBalance = Lesson7.StringArrayToFloatArray(firstAccount[1..]);
+
+            var names = new string[] {firstAccount[0]};
+            var highestBalance = Lesson7.MaxFloat(firstBalance);
+
+            for (int i = 1; i < peopleAndBalances.Length; i++)
+            {
+                var currentAccount = peopleAndBalances[i].Split(", ");
+                var currentBalance = Lesson7.StringArrayToFloatArray(currentAccount[1..]);
+
+                var currentName = currentAccount[0];
+                var currentHighest = Lesson7.MaxFloat(currentBalance);
+
+                // if same as others add it to the list
+                if (currentHighest == highestBalance)
+                {
+                    names = Lesson7.AppendString(names, currentName);
+                }
+
+                if (currentHighest > highestBalance)
+                {
+                    names = new string[] {currentName};
+                    highestBalance = currentHighest;
+                }
+            }
+
+            string returnNames;
```

Comment:
Perfect case for splitting to a function.

Reply:
Created FormatArrayToString function.